backend/src/ml_space_lambda/utils/dict_utils.py

```diff
@@ -0,0 +1,27 @@
+def filter_dict(input: dict, allowed_keys: list[str]):
+    """
+    Returns a copy of input with keys not in allowed_keys removed
+
+    Args:
+        input (dict): A dict to filter
+        allowed_keys (list[str]): A list of allowed keys
+
+    Returns:
+        dict: A copy of input with only keys specified in allowed_keys
+    """
+    return {key: input.get(key) for key in allowed_keys if input.get(key) is not None}
+
+
+def map_dict_keys(input: dict, mapping: dict):
```

Comment:
Kyle originally asked me to break it out so it wasn't just inside the `datasets/lambda_functions.py` but much like the other utility since it is only now being used once I don't necessarily need it to be its down function.

Reply:
Keeping as a util is fine. I would like to rename as "map" is an overloaded term. Also second parameter name should be changed to "nameMappings" or something equivalent that more clearly indicates that these are names being replaced.